vault/core_metrics_test.go

```diff
@@ -78,15 +79,23 @@ func TestCoreMetrics_KvSecretGauge(t *testing.T) {
 		}
 	}
 	for _, p := range v2secrets {
-		req := logical.TestRequest(t, logical.CreateOperation, p)
-		req.Data["data"] = map[string]interface{}{"foo": "bar"}
-		req.ClientToken = root
-		resp, err := core.HandleRequest(ctx, req)
-		if err != nil {
-			t.Fatalf("err: %v", err)
-		}
-		if resp.Error() != nil {
-			t.Fatalf("bad: %#v", resp)
+		for {
+			req := logical.TestRequest(t, logical.CreateOperation, p)
+			req.Data["data"] = map[string]interface{}{"foo": "bar"}
+			req.ClientToken = root
+			resp, err := core.HandleRequest(ctx, req)
+			if err != nil {
+				if errors.Is(err, logical.ErrInvalidRequest) {
+					// Handle scenario where KVv2 upgrade is ongoing
+					time.Sleep(100 * time.Millisecond)
+					continue
```

Comment:
I'd feel better if there were a maximum number of retries.

Reply:
Done.